tests/testthat/test_exponential_decay_corr_func.R

```diff
@@ -0,0 +1,38 @@
+test_that(
+  "Test Exponential Decay Corr Func in Stan agrees with R",
+  {
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns
+
+    corr_func_params <- list(
+      dist_matrix = as.matrix(dist(
+        data.frame(
+          x = runif(5, 0, 100),
+          y = runif(5, 0, 100)
+        ),
+        diag = TRUE,
+        upper = TRUE
+      )),
+      phi = 25,
+      l = 1
+    )
+
+    exp_decay_corr_func_stan <-
+      space_model_fxns$functions$exponential_decay_corr_func(
+        corr_func_params$dist_matrix,
+        corr_func_params$phi,
+        corr_func_params$l
+      )
+    exp_decay_corr_func_r <- exponential_decay_corr_func_r(
+      corr_func_params
+    )
+    # both functions return correlation matrix using an
+    # exponential decay correlation function
+
+    testthat::expect_equal(
```

Comment:
You may want `expect_almost_equal()` here to make the test robust to potential minor differences in float arithmetic

Reply:
@dylanhmorris which library is that in, it doesn't seem to be in the testthat library.  expect_equal() does have a tolerance option if that also works.

Reply:
My mistake, thanks @cbernalz. I forgot that in `testthat`, [`expect_equal()` is the name of the float-tolerant expectation and `expect_identical()` is the name of the stricter expectation](https://testthat.r-lib.org/reference/equality-expectations.html).